### ISSUE TITLE
`fix-smartquotes` modifying non-quote characters

### ISSUE THREAD

**Issue by sco1** [sirosen/texthooks#40]
I haven't poked around to figure out why this is happening, but it appears that the smartquotes fixer is making changes to characters that aren't smartquotes.

I was running as a pre-commit hook, but this should reproduce with Python `3.11.0` & texthooks `v0.4.0`:
```bash
$ echo "“foo–bar”" > foo.txt
$ fix-smartquotes ./foo.txt 
Changes were made in these files:
  ←[0;33m./foo.txt←[0m
$ cat foo.txt
"foo□bar"
```

Follow-on invocations now raise:
```bash
$ fix-smartquotes ./foo.txt 
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "~\sco1\tmp\.venv\Scripts\fix-smartquotes.exe\__main__.py", line 7, in <module>
  File "~\sco1\tmp\.venv\Lib\site-packages\texthooks\fix_smartquotes.py", line 131, in main
    changes = do_all_replacements(
              ^^^^^^^^^^^^^^^^^^^^
  File "~\sco1\tmp\.venv\Lib\site-packages\texthooks\fix_smartquotes.py", line 77, in do_all_replacements
    recorder.run_line_fixer(line_fixer, fn)
  File "~\sco1\tmp\.venv\Lib\site-packages\texthooks\_recorders.py", line 71, in run_line_fixer
    content = _compat_readlines(filename)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\sco1\tmp\.venv\Lib\site-packages\texthooks\_recorders.py", line 39, in _compat_readlines
    return f.readlines()
           ^^^^^^^^^^^^^
  File "<frozen codecs>", line 322, in decode
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x96 in position 4: invalid start byte
```

Tangentially related, would it be possible to include some debug information to help identify which file(s) are failing?

**Comment by sirosen** [comment]
Thanks for including a complete MRE! Huge help to have that; I agree that it's wrong. I'll take a look at this as soon as I'm able.

For added debug info, would adding a `-v/--verbose` flag which prints each filename before it's checked/fixed, would that be acceptable? Then you could opt-in to `-v` but not everyone needs to have it on.
I can also try to catch errors and handle them more intelligently, but there's so much more risk that I miss a case that I'd rather let users ask the hook to print filenames with `-v`.

**Comment by sco1** [comment]
A verbose flag would be fine, it was just tough trying to find the needle in the haystack :)

**Comment by sirosen** [comment]
I think I have this fixed.

The encoding for files was not being set on writes consistently with the way it was detected for reads. The regression test I added failed on Windows without the fix, and is now passing.

Two quick notes:
- I'll release shortly, so that you can try this out and see if it works for you
- First I'm going to see how quickly I can get a `-v/--verbose` flag added

**Comment by sirosen** [comment]
I've just released v0.5.0 with a fix for this and `--verbose`. :confetti_ball: 

Please let me know if it doesn't resolve the issue, but I feel pretty good about the fix.

_If_ it doesn't work, some things that would be helpful to get some more details.
In particular:
- This is happening on Windows? (That's where I was able to reproduce.)
- What is the output of `python -c 'import sys; print(sys.getfilesystemencoding()); print(sys.getdefaultencoding())'`

**Comment by sco1** [comment]
Thanks for the quick fix! Looks good to me on MacOS, I won't be back to a windows machine until next weekend; will let you know if I see any issue.
